src/components/Tasks.js

```diff
@@ -15,74 +16,65 @@ export const Tasks = () => {
 
     const [tempItem, setTempItem] = useState(initial);
 
-    const getItems = () => {
-        fetch(uri)
-        .then(res => res.json())
-        .then(
-            result => {
-                setIsLoaded(true);
-                setTodoItems(result);
-            }, error => {
-                setIsLoaded(true);
-                setError(error);
-            }
-        );
+    const getItems = async () => {
+        //setIsLoaded(false);
+        try {
+            const result = await getItemsApi();
+            setTodoItems(result);
+        } catch (e) {
+            setError(`Error getting items: ${e}`);
+        }
+        setIsLoaded(true);
     };
 
-    const addItem = () => {
+    const addItem = async () => {
         const item = {
             isComplete: false,
             name: newTodoName.trim()
         };
 
-        fetch(uri, {
-            method: 'POST',
-            headers: {
-                'Accept': 'application/json',
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify(item)
-        })
-        .then(response => response.json())
-        .then(() => {
+        try {
+            await addItemApi(item);
             getItems();
+            // setTodoItems([...todoItems, {
+            //     id: item.id,
+            //     ...item
+            // }])
             setNewTodoName('');
-            }, error => setError(error)
-        );
+        } catch (e) {
+            setError(`Error adding item: ${e}`);
+        }
     };
 
-    const deleteItem = id => {
-        fetch(`${uri}/${id}`, {
-            method: 'DELETE'
-        })
-        .then(() => getItems(), error => setError(error));
+    const deleteItem = async (id) => {
+        try {
+            await deleteItemApi(id);
+            getItems();
```

Comment:
Same here, easier to just update state by removing the item from state using the passed in id (instead of making another call)